### PR TITLE
feat: add build scripts for core data schema

### LINKED ISSUE #3: Scripts for creating the core model
Create SQL scripts for creating tables based on the domain model and agreed upon data model for core data

Entities that should be included:
- `media`
  - `series`
  - `season`
  - `episode`
  - `movie`
- `release`
- `person`
- `crew_member`
- `promotional_media`
- `production_company`
- `score`

Simple values (only contains names)
- `genre`
- `country`
- `language`

### LINKED ISSUE #7: ER-diagram (CORE)
Generate er diagram for core data model

### PR DESCRIPTION
**Pull request by sofushn**
This PR is for adding build script for the core data model of the movie database.

Features:
- Added: entity relationship diagram for core data model
- Added: Build SQL script for all tables in the core data model
  - `DROP` tables before adding them to allow for multiple executions
  - `CREATE` tables corresponding to each entity in domain model
- Added: new entity **Cast Member** which is a [specialization](https://www.geeksforgeeks.org/generalization-specialization-and-aggregation-in-er-model/) of **Crew Member** but is always an actor.
- Added: Description to **Production Company**
- Changed: **Country**, **Language** and **Job Category** to their own entities so they can be reused.
- Changed: relation for **Promotional Media** to **Release** instead **Media**.
- Changed: domain model in response to changes.

closing #3 and #7

### CODE REVIEW

File: src/B2_build_movie_db.sql [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    spoken_language_id  INTEGER     PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    release_id          INTEGER     NOT NULL REFERENCES release(release_id),
+    language_code       VARCHAR(2)  NOT NULL REFERENCES "language"(code)
+);
+
+CREATE TABLE promotional_media (
+    promotional_media_id    INTEGER         PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    release_id              INTEGER         NOT NULL REFERENCES release(release_id),
+    "type"                  VARCHAR(50)     NOT NULL,
+    "uri"                   VARCHAR(255)    NOT NULL
```
Comment (JuliusKryger) [inline src/B2_build_movie_db.sql:124]: Couldn't we have a scenario where a promotional media, do not have an uri? Or do we want to force that a promotional media should always have an uri?

File: domain_model.drawio.png
Comment (JuliusKryger) [inline domain_model.drawio.png:1]: We can safely remove the plan_to_watch on the framework side?
 In the framework, the search_history actually have a reference to which user did the search.

**Review by JuliusKryger — CHANGES_REQUESTED** [review]

File: src/B2_build_movie_db.sql (hunk 1 above)
Comment (sofushn) [reply src/B2_build_movie_db.sql:124]: Isn't it kinda the point of promotional media? If it has no `uri` why would it even exist?

Comment (JuliusKryger) [reply src/B2_build_movie_db.sql:124]: Well I suppose it is going to depend on the `type` of promotional media that we decide to support. I am okay with us deciding that promotional media, must always have an website in order to be considered promotional media in our system. Just wanted to make sure it was not an oversight. :)

**Review by JuliusKryger — APPROVED** [review]

**Pull request merged**